docs/en/reference/architecture.rst

```diff
@@ -43,8 +43,8 @@ interfaces:
 -  ``Doctrine\DBAL\Driver\Statement``
 -  ``Doctrine\DBAL\Driver\Result``
 
-Middlewares
------------
+Middleware
+----------
 
 A middleware sits in the middle between the wrapper components and the driver.
```

Comment:
If middleware is not countable, is there something like _a_ middleware? The way you use the term in the `Configuration` class now, middleware is the whole layer between the driver an the wrapper. I think, we need to find a different term for the thing that we create through implementing the `Middleware` interface.